GraalVM Reflection guide (#1346)
micronaut-projects/micronaut-guides: guides/micronaut-graalvm-reflection/micronaut-graalvm-reflection.adoc

common:header.adoc[]

In this tutorial, you are going to learn how to use https://www.graalvm.org/latest/reference-manual/native-image/dynamic-features/Reflection/[Reflection in Native Image] within a Micronaut application.

____
Java reflection support (the java.lang.reflect.* API) enables Java code to examine its own classes, methods, fields and their properties at run time.

> Native Image has partial support for reflection and needs to know ahead-of-time the reflectively accessed program elements.

Reviewer: ```suggestion
Native Image has partial support for reflection but needs to know ahead-of-time the reflectively accessed program elements.
```

____

Micronaut Framework internals do not use reflection. However, you may want to use reflection in your application or use a Java library that uses reflection.

> In this tutorial, you will create a Micronaut application that uses reflection, and you will see how to configure it to work with Native Image.

Reviewer: ```suggestion
This tutorial will create a Micronaut application that uses reflection, and then show different techniques to configure it to work with Native Image.
```

common:header.adoc[]

In this tutorial, you are going to learn how to use https://www.graalvm.org/latest/reference-manual/native-image/dynamic-features/Reflection/[Reflection in Native Image] within a Micronaut application.

____
Java reflection support (the java.lang.reflect.* API) enables Java code to examine its own classes, methods, fields and their properties at run time.

> Native Image has partial support for reflection but needs to know ahead-of-time the reflectively accessed program elements.

Reviewer 2: Native Image fully supports reflection it just needs to be configured

Author: Please, note it is a quote from the GraalVM website: 

https://www.graalvm.org/latest/reference-manual/native-image/dynamic-features/Reflection

<img width="975" alt="CleanShot 2023-09-26 at 11 18 19@2x" src="https://github.com/micronaut-projects/micronaut-guides/assets/864788/4d043bea-8079-4736-a6e6-084c6bbaaca2">

Please, give me a suggestion of what we should write instead?

Reviewer 2: ok probably fine then

Author: I talked to Fabio yesterday and we updated it to: https://github.com/micronaut-projects/micronaut-guides/pull/1346/commits/4dd2bcf8c3e9bd06d6c991a967b2fed204253afa

It is going to be updated in the GraalVM website as well.

> Native Image has partial support for reflection but needs to know ahead-of-time the reflectively accessed program elements.

Author: ```suggestion
Native Image supports reflection but needs to know ahead-of-time the reflectively accessed program elements.
```
